google/resumable_media/_upload.py

```diff
@@ -468,9 +468,10 @@ def _prepare_initiate_request(
         parse_result = urllib.parse.urlparse(self.upload_url)
         parsed_query = urllib.parse.parse_qs(parse_result.query)
         if "x-goog-signature" in parsed_query or "X-Goog-Signature" in parsed_query:
-            headers = {_CONTENT_TYPE_HEADER: content_type}
+            headers = {**self._headers, _CONTENT_TYPE_HEADER: content_type}
```

Comment:
Are there tests that you can update as well?

Reply:
Done. There was already a test that ensured they were merged together, what we changed was putting these ones first, to ensure the content_type and x-upload-content-type described here took priority. And so I just adjusted the prepare_initiate_request_with_headers by adding a content-type and ensuring it was overwritten.